psycop/common/model_training_v2/trainer/base_trainer.py

```diff
@@ -1,21 +1,38 @@
 # Implement this object for cross-validation, split-validation
+import pickle
+import tempfile
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Protocol, runtime_checkable
+from pathlib import Path
 
 import polars as pl
 
 from psycop.common.model_training_v2.trainer.task.base_metric import (
     CalculatedMetric,
 )
 
+from ..loggers.base_logger import BaselineLogger
+from .task.base_task import BaselineTask
+
 
 @dataclass(frozen=True)
 class TrainingResult:
     metric: CalculatedMetric
     df: pl.DataFrame
 
 
-@runtime_checkable
-class BaselineTrainer(Protocol):
+class BaselineTrainer(ABC):
+    logger: BaselineLogger
+    task: BaselineTask
+
+    @abstractmethod
     def train(self) -> TrainingResult:
         ...
+
+    def _log_sklearn_pipe(self) -> None:
+        with tempfile.NamedTemporaryFile(prefix="sklearn_pipe", suffix=".pkl") as f:
+            pickle.dump(self.task.pipe, f)
```

Comment:
This will not log the sklearn pipeline but the our task pipeline (which has an sklearn_pipe) attribute. We should probably just log the sklearn pipeline

Reply:
Completely agree, fixing.